### ISSUE TITLE
Server crashed, htmlPlugin.options is undefined

### ISSUE THREAD

**Issue by Akiyamka** [react-cosmos/react-cosmos#1285]
### What's wrong?
Webpack 5.18.0 project.
Install 
```
[Cosmos] Using default cosmos config
[Cosmos] See you at http://localhost:5000
[Cosmos] Using default webpack config
[Cosmos] Server crashed...

  (╯°□°)╯︵ ┻━┻

TypeError: Cannot read property 'filename' of undefined
```
[Place where error throw](https://github.com/react-cosmos/react-cosmos/blob/dbe83a74f53de7ae3dd61f822aec56e673f725b9/packages/react-cosmos/src/plugins/webpack/webpackConfig/htmlPlugin.ts#L69)

### Steps to reproduce
1. Create new project with with next webpack [config](https://gist.github.com/Akiyamka/e6cca2ccaa0f3f8384a989d4743acdf3)
Few stupid simple react components <App /> <HelloWorld /> and <Button />, check it work as expected.
2.
```
npm i --D react-cosmos
```
3. add 
```
    "cosmos": "cosmos",
    "cosmos:export": "cosmos-export"
 ```
 to package.json scripts
4.
```
npm run cosmos
```

### Screenshots

![image](https://user-images.githubusercontent.com/6587585/106128186-3e09e780-6170-11eb-98f2-dd231052e98e.png)

### User info

- Server terminal output: 
```
npm run cosmos

> landing@0.0.1 cosmos .../landing
> cosmos

[Cosmos] Using default cosmos config
[Cosmos] See you at http://localhost:5000
[Cosmos] Using default webpack config
[Cosmos] Server crashed...

  (╯°□°)╯︵ ┻━┻

TypeError: Cannot read property 'filename' of undefined
    at isIndexHtmlWebpackPlugin (.../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/htmlPlugin.js:75:37)
    at changeHtmlPluginFilename (.../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/htmlPlugin.js:67:8)
    at .../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/htmlPlugin.js:42:44
    at Array.map (<anonymous>)
    at ensureHtmlWebackPlugin (.../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/htmlPlugin.js:41:20)
    at getPlugins (.../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/devServer.js:137:49)
    at _callee$ (.../landing/node_modules/react-cosmos/dist/plugins/webpack/webpackConfig/devServer.js:65:24)
    at tryCatch (.../landing/node_modules/regenerator-runtime/runtime.js:63:40)
    at Generator.invoke [as _invoke] (.../landing/node_modules/regenerator-runtime/runtime.js:293:22)
    at Generator.next (.../landing/node_modules/regenerator-runtime/runtime.js:118:21)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! landing@0.0.1 cosmos: `cosmos`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the landing@0.0.1 cosmos script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     .../.npm/_logs/2021-01-28T10_31_36_997Z-debug.log
```

- Browser console output: null
- Cosmos version: 5.5.3
- Cosmos config: ["No config is required to star" ](https://github.com/react-cosmos/react-cosmos/tree/master/docs#config)
-  webpack version: 5.18
-  webpack [config](https://gist.github.com/Akiyamka/e6cca2ccaa0f3f8384a989d4743acdf3)

**Comment by skidding** [comment]
Hi @Akiyamka,

Recently published an alpha version that's meant to work with webpack 5. Can you please try `react-cosmos@5.6.0-alpha.0`?

Moreover, your error makes me think that you're using [html-webpack-plugin](https://www.npmjs.com/package/html-webpack-plugin) version 5 (beta)? If that is the case, can you install `html-webpack-plugin@4.5.1` (latest) instead?

**Comment by Akiyamka** [comment]
@skidding, thanks for the quick reply!
> Moreover, your error makes me think that you're using html-webpack-plugin version 5 (beta)
You right `5.0.0-beta.6`
>  If that is the case, can you install html-webpack-plugin@4.5.1 (latest) instead?
Yes, I can, but I wouldn't like to do it since 4.5.1 version uses a deprecated API
> Can you please try react-cosmos@5.6.0-alpha.0
Of course, I'll try, if not helps I will to downgrade the version of the html plugin

**Comment by Akiyamka** [comment]
@skidding after downgrade to html-webpack-plugin@4.5.1 react-cosmos work correct
Just in case - `console.log(plugins)` from [here](https://github.com/react-cosmos/react-cosmos/blob/dbe83a74f53de7ae3dd61f822aec56e673f725b9/packages/react-cosmos/src/plugins/webpack/webpackConfig/devServer.ts#L98) with `html-webpack-plugin@5.0.0-beta.6`

```
[
  HtmlWebpackPlugin {
    userOptions: { title: 'React Cosmos', filename: '_renderer.html' },
    version: 5
  },
  DefinePlugin {
    definitions: { __DEV__: 'true', 'process.env': [Object] }
  },
  NoEmitOnErrorsPlugin {},
  HotModuleReplacementPlugin { options: {} }
]
```
options become userOptions ?

**Comment by skidding** [comment]
> @skidding after downgrade to html-webpack-plugin@4.5.1 react-cosmos work correct.

Good to hear. Sorry about the temporary deprecation warning.

> options become userOptions

Indeed. I almost shipped this change, but then realised it will break for everyone not using html-webpack-plugin 5, which is still beta! :)

I could publish RC v6, but it seems over the top to create a major version for a html webpack plugin upgrade. The real solution is to support both versions (welcome to the world of tool making...) and use options/userOptions based on introspection of the html webpack plugin version. I can provide pointers if you're interested to give a hand with this PR ;).

**Comment by skidding** [comment]
@Akiyamka Thanks for your contribution!

`react-cosmos@5.6.0-alpha.1` contains your patch. Please try it out and let me know if it works as expected.

**Comment by supermihi** [comment]
FYI, I had the same problem and `react-cosmos@5.6.0-alpha.2` solved it (using `html-webpack-plugin@5.0.0`). Thanks!
